flake.nix

```diff
@@ -1,98 +1,30 @@
-  inputs.crate2nix = { url = "github:kolloch/crate2nix"; flake = false; };
-  inputs.mozilla = { url = "github:mozilla/nixpkgs-mozilla"; flake = false; };
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs =
-    { self
-    , nixpkgs
-    , mozilla
-    , ...
-    } @ inputs:
+  outputs = { self, nixpkgs, flake-utils }: flake-utils.lib.eachDefaultSystem (system:
     let
-      nameValuePair = name: value: { inherit name value; };
-      genAttrs = names: f: builtins.listToAttrs (map (n: nameValuePair n (f n)) names);
-      allSystems = [ "x86_64-linux" "aarch64-linux" "i686-linux" "x86_64-darwin" ];
-
-      # rustOverlay = final: prev:
-      #   let
-      #     rustChannel = prev.rustChannelOf {
-      #       channel = "1.47.0";
-      #       sha256 = "1hkisci4as93hx8ybf13bmxkj9jsvd4a9ilvjmw6n64w4jkc1nk9";
-      #     };
-      #   in
-      #   {
-      #     inherit rustChannel;
-      #     rustc = rustChannel.rust;
-      #     cargo = rustChannel.rust;
-      #   };
-
-      forAllSystems = f: genAttrs allSystems (system: f {
-        inherit system;
-        pkgs = import nixpkgs {
-          inherit system;
-          # overlays = [
-          #   (import "${mozilla}/rust-overlay.nix")
-          #   rustOverlay
-          # ];
-        };
-      });
-
-      crate2nix = forAllSystems ({ system, ... }:
-        let
-          sources = import "${inputs.crate2nix}/nix/sources.nix";
-          pkgs = import sources.nixpkgs { inherit system; };
-          cargoNix = import "${inputs.crate2nix}/crate2nix/Cargo.nix" {
-            inherit pkgs;
-            defaultCrateOverrides = pkgs.defaultCrateOverrides // {
-              cssparser-macros = { ... }: {
-                buildInputs = with pkgs;
-                  lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
-              };
-            };
-          };
-        in
-        cargoNix.rootCrate.build);
+      pkgs = nixpkgs.legacyPackages.${system};
     in
     {
-      # for use with update.sh script
-      inputs = builtins.removeAttrs inputs [ "self" ];
+      packages.agenix-cli = pkgs.callPackage
+        ({ lib, rustPlatform }:
 
-      devShell = forAllSystems ({ system, pkgs, ... }:
-        pkgs.mkShell {
-          name = "agenix";
+          rustPlatform.buildRustPackage {
+            pname = "agenix-cli";
+            version = (lib.importTOML ./Cargo.toml).package.version;
 
-          # Eventually crate2nix will provide a devShell that includes transitive dependencies for us.
-          # https://github.com/kolloch/crate2nix/issues/111
-          buildInputs = with pkgs; [
-            # rustChannel.rust provides tools like clippy, rustfmt, cargo,
-            # rust-analyzer, rustc, and more.
-            # (rustChannel.rust.override { extensions = [ "rust-src" ]; })
-            crate2nix.${system}
-          ];
-        });
+            src = ./.;
```

Comment:
I think I prefer having `src = self;` so that when changing things outside the flake (e.g. the gitignore'd `target` build output directory), the package doesn't have to be rebuilt in its entirety.

Reply:
Again, I didn't even know this was possible, whoa. Thank you.